internal/experiment/generation/generation.go

```diff
@@ -111,3 +123,51 @@ func loadApplicationData(app *redskyappsv1alpha1.Application, src string) ([]byt
 
 	return ioutil.ReadFile(dst)
 }
+
+var metricSelectorPattern = regexp.MustCompile(`([a-zA-Z_][a-zA-Z0-9_]*)(=|!=|=~|!~)"([a-zA-Z0-9\-|]+)"`)
+
+// convertPrometheusSelector converts a Prometheus metric selector to a Kubernetes
+// label selector. This is necessary because objectives like "Requests" define their
+// "selector" as a Prometheus selector: in order for that to work with a metric
+// with `type: kubernetes` (or `type: ""`), we must first convert it over.
+func convertPrometheusSelector(metricSelector string) (*metav1.LabelSelector, error) {
```

Comment:
is this replacing #250?

Reply:
No, this is because there are two different types of selectors and we need convert between them. When you configure objectives you use the field named `metricsSelector` which corresponds to the Prometheus version of a label selector. Almost everywhere else you use a field named `labelSelector` which corresponds to a Kubernetes version of label selector. In this particular case, we are given a Prometheus label selector (which would typically just be injected into a Prometheus query) but we need to use it in a context where we are selecting Kubernetes resources (the `PodList` request to the Kube API), so we must convert the selectors (to the best of our ability, Prometheus selectors are more expressive/less efficient). The point of #250 is more to reduce the limitations we have artificially enforced on the syntax of the Prometheus label selectors.